lib/hrr_rb_ssh/transport/server_host_key_algorithm/ecdsa_sha2_nistp256.rb

```diff
@@ -7,7 +7,7 @@ class EcdsaSha2Nistp256 < ServerHostKeyAlgorithm
         NAME = 'ecdsa-sha2-nistp256'
         PREFERENCE = 30
         IDENTIFIER = 'nistp256'
-        SECRET_KEY = OpenSSL::PKey::EC.new('prime256v1').generate_key.to_pem
+        SECRET_KEY = OpenSSL::PKey::EC.generate('prime256v1').to_pem
```

Comment:
This is the actual change needed to fix https://github.com/rapid7/metasploit-framework/issues/16767#issuecomment-1188563161 :+1: